src/Microsoft.Health.Fhir.Api.UnitTests/Controllers/ExportControllerTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Net;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Api.Controllers;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Context;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Api.UnitTests.Controllers
+{
+    public class ExportControllerTests
+    {
+        [Fact]
+        public void GivenAnExportRequest_WhenDisabled_ThenBadRequestResponseShouldBeReturned()
+        {
+            var exportController = GetController(new ExportConfiguration() { Enabled = false });
```

Comment:
You should consider move the creation logic into one location like in constructor and save the controller as an instance variable as opposed to create it in every single test.

If you ever change the signature of the constructor, it's one place to fix as opposed to multiple places.

Reply:
I agree with what you said. But in this scenario I need to inject a different `ExportConfiguration` depending on the test. I can't see a way to change the exportconfiguration of an existing controller unless I change the access modifier.

I can create two instances of the controller for these tests (one with config enabled and another with disabled). We would also need to create new instances for each parameter we add to `ExportConfiguration`. But that would restrict future changes to just the `ExportControllerTests()` ctor where we will have one instance for each config we are testing.